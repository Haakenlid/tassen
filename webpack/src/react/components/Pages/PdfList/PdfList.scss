@import "styles/_common.scss";

.PdfList {
  display: grid;
  /* prettier-ignore */
  grid-template-columns: 1fr 1fr;
  grid-gap: 1rem;

  @include size(xs) {
    grid-template-columns: repeat(auto-fill, minmax(12rem, 1fr));
    grid-gap: 2rem;
  }

  .Issue {
    position: relative;
    display: flex;
    flex-direction: column;
    &.two {
      grid-column-end: span 2;
      .Faximile {
        &:nth-child(1) {
          z-index: 10;
          transform: scale(0.95) rotateZ(0deg) translateX(15%);
        }
        &:nth-child(2) {
          transform: scale(0.95) rotateZ(5deg) translateX(-15%);
        }
      }
    }
    &:hover {
      .text {
        color: $dark-grey;
      }
      .pdf {
        .Faximile {
          transform: scale(1);
        }
      }
    }
    .text {
      border-bottom: 1px solid #eee;
      margin-bottom: 0.5em;
      .name {
        font-weight: bold;
      }
      @extend %smaller;
      // font-size: 0.8em;
      display: flex;
      flex-direction: column;
      @include size(xs) {
        flex-direction: row;
      }
      align-items: center;
      justify-content: space-around;
    }
    .pdf {
      display: grid;
      grid-auto-flow: column;
      grid-gap: 2rem;
      flex: 1;
      .Faximile {
        background: #eee;
        @extend %faximile;
        transition: transform 300ms ease;
        transition-delay: 200ms;
        transform: scale(0.95);
        padding-top: 143.5%;
        img {
          top: 0;
          position: absolute;
          width: 100%;
        }
      }
    }
  }
}
