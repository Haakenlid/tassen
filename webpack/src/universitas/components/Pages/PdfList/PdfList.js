import { formatDate } from 'utils/text'
import cx from 'classnames'

const filename = R.replace(/.*\//, '')

const Faximile = ({ cover, pages, pdf }) => (
  <div className="Faximile" key={cover}>
    <a href={pdf}>
      <img alt={filename(pdf)} src={cover} />
    </a>
  </div>
)

const sortPdfs = R.propSatisfies(R.test(/_mag/), 'pdf')

const Issue = ({ publication_date, issue_name, pdfs }) => (
  <div className={cx('Issue', { two: pdfs.length == 2 })} key={issue_name}>
    <div className="text">
      <div className="name">{issue_name}</div>
      <div className="date">{formatDate(publication_date, 'dddd D. MMMM')}</div>
    </div>
    <div className="pdf">
      {R.pipe(R.sortBy(sortPdfs), R.map(Faximile))(pdfs)}
    </div>
  </div>
)

const PdfList = ({ issues = [] }) => (
  <section className="PdfList">
    {R.pipe(
      R.filter(R.propSatisfies(R.length, 'pdfs')),
      R.sortBy(R.prop('publication_date')),
      R.reverse,
      R.map(Issue),
    )(issues)}
  </section>
)

export default PdfList
