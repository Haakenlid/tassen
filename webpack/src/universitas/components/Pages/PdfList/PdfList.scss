@import "styles/_common.scss";

.PdfList {
  display: grid;
  /* prettier-ignore */
  grid-template-columns: 1fr 1fr;
  grid-gap: 1rem;

  @include size(xs) {
    grid-template-columns: repeat(auto-fill, minmax(12rem, 1fr));
    grid-gap: 2rem;
  }

  .Issue {
    &.two {
      grid-column-end: span 2;
      .pdf {
        display: flex;
        .Faximile {
          padding-top: calc(143.5% * 0.5 - 1.35rem);
          &:nth-child(1) {
            margin-right: 1rem;
            z-index: 10;
            transform: scale(0.95) rotateZ(0deg) translateX(15%);
          }
          &:nth-child(2) {
            margin-left: 1rem;
            transform: scale(0.95) rotateZ(5deg) translateX(-15%);
          }
        }
      }
    }
    &:hover {
      .text {
        color: $dark-grey;
      }
    }
    .text {
      border-bottom: 1px solid #eee;
      margin-bottom: 0.5em;
      .name {
        font-weight: bold;
      }
      @extend %smaller;
      // font-size: 0.8em;
      display: flex;
      flex-direction: column;
      @include size(xs) {
        flex-direction: row;
      }
      align-items: center;
      justify-content: space-around;
    }
    .pdf {
      flex: 1;
      .Faximile {
        background: #eee;
        @extend %faximile;
        transition: transform 300ms ease;
        transition-delay: 100ms;
        transform: scale(0.95);
        &:hover {
          transform: scale(1) !important;
        }
        width: 100%;
        padding-top: 143.5%;
        img {
          top: 0;
          position: absolute;
          width: 100%;
        }
      }
    }
  }
}
