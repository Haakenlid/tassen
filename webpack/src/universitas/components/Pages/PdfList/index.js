import PdfList from './PdfList.js'
import './PdfList.scss'
export default PdfList
