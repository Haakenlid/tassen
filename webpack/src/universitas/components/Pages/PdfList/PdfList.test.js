import PdfList from './index'
import renderer from 'react-test-renderer'

describe('PdfList', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<PdfList />).toJSON()
    expect(tree).toEqual(expect.any(Object))
    expect(tree).toMatchSnapshot()
  })
})
