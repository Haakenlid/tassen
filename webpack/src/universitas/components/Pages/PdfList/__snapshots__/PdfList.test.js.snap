// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PdfList renders correctly 1`] = `
<section
  className="PdfList"
/>
`;
